packages/teams-js/src/public/thirdPartyCloudStorage.ts

```diff
@@ -0,0 +1,201 @@
+ * @beta
+ */
+export namespace thirdPartyCloudStorage {
+  /** Get context callback function type */
+  const files: FilesFor3PApps[] = [];
+  let helper: AttachmentListHelper = {
+    fileType: '',
+    assembleAttachment: [],
+  };
+
+  /**
+   * Object used to represent a file
+   * @beta
+   *
+   */
+  export interface FilesFor3PApps extends Blob {
+    /**
+     * A number that represents the number of milliseconds since the Unix epoch
+     */
+    lastModified: number;
+    /**
+     * Name of the file
+     */
+    name: string;
+    /**
+     * file type
+     */
+    type: string;
+    /**
+     * A string containing the path of the file relative to the ancestor directory the user selected
+     */
+    webkitRelativePath?: string;
+  }
+
+  /**
+   * File chunks an output of getDragAndDropFiles API from platform
+   */
+  export interface FileChunk {
+    /**
+     * Base 64 data for the requested uri
+     */
+    chunk: string;
+    /**
+     * chunk sequence number
+     */
+    chunkSequence: number;
+  }
+
+  /**
+   * Output of getDragAndDropFiles API from platform
+   */
+  export interface FileResult {
+    /**
+     * Error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type which is getting recieved
+     */
+    fileType: string;
+    /**
+     * Indicates whether this file is the last one in a sequence.
+     */
+    isLastFile: boolean;
+    /**
+     * The name of the file.
+     */
+    fileName: string;
+  }
+
+  /**
+   * Helper class for assembling files
+   */
+  export interface AttachmentListHelper {
+    /** A string representing the MIME type of the file */
+    fileType: string;
+    /** An array of {@link media.AssembleAttachment | AssembleAttachment} objects representing the media files to be sent as attachment */
+    assembleAttachment: media.AssembleAttachment[];
+  }
+
+  /**
+   * Defines the Callback function received from Third Party App
+   */
+  export interface ThirdPartyAppCallback {
+    /** Callback from third party app */
+    (files: FilesFor3PApps[], error?: SdkError): void;
+  }
+
+  let callback: ThirdPartyAppCallback | null = null;
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param {string} dragAndDropInput - The input related to the drag-and-drop operation.
+   * @param {ThirdPartyAppCallback} thirdPartycallback - callback
+   *   A callback function to handle the result of the operation
+   */
+  export function getDragAndDropFiles(dragAndDropInput: string, thirdPartycallback: ThirdPartyAppCallback): void {
+    if (!thirdPartycallback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+    callback = thirdPartycallback;
```

Comment:
all of your namespace-level variables should be cleared out once an entire image / set of images is retrieves (callback, handler, files) so that they don't contain incorrect data when a second image is drag and dropped.

Reply:
We will receive callback from 3P app once and send the array of all the dropped files suing that callback.
Flow is:
We get callback and thread id from 3p app,
Based on thread id we get the files from TMP
Once all files are received, we will send the array of files to 3P app using their callback,

Reply:
I'm not sure what your reply means in relation to my question, but it does look like you are resetting the handler and files array now. I don't think you're resetting the callback, but since there can only ever be one callback as you said above that should be okay.